### ISSUE TITLE
Error loading module error_Logger

### ISSUE THREAD

**Issue by lukaszsamson** [elixir-lang/elixir#12604]
### Elixir and Erlang/OTP versions

Erlang/OTP 25 [erts-13.2.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]

Elixir 1.15.0-rc.0 (d673191) (compiled with Erlang/OTP 25)

### Operating system

macOS

### Current behavior

Some dependencies output error during build with MIX_ENV=test
```
14:56:41.065 [error] beam/beam_load.c(180): Error loading module error_Logger:
  module name in object code is error_logger

14:56:41.065 [error] Loading of /Users/myuser/.asdf/installs/erlang/25.3.2/lib/kernel-8.5.4/ebin/error_Logger.beam failed: :badfile

warning: :error_Logger.warninging_msg/2 is undefined (module :error_Logger is not available or is yet to be defined)
  lib/phoenix/channel/server.ex:541: Phoenix.Channel.Server.warn_unexpected_msg/3
```
gen_stage v1.2.1 (latest)
phoenix v1.6.16

### Expected behavior

No error?

**Comment by josevalim** [comment]
This doesn't seem to be an Elixir issue. Phoenix is correctly calling a function here: https://github.com/phoenixframework/phoenix/blob/v1.6.16/lib/phoenix/channel/server.ex#L551

Perhaps the dependency was changed somehow?

Can you reproduce it?

**Comment by wojtekmach** [comment]
I created a new project with:

```elixir
  defp deps do
    [
      {:gen_stage, "1.2.1"},
      {:phoenix, "1.6.16"}
    ]
  end
```

and couldn't reproduce this. I'm on a case-insensitive file-system at the moment.

Any chance you can reproduce this with a Mix.install?

**Comment by josevalim** [comment]
Yeah, I checked the dependency. I believe this is a regex gone wrong. All uses of `Logger.warn` (case insensitive) were replaced by `Logger.warning` :)

**Comment by lukaszsamson** [comment]
> Yeah, I checked the dependency. I believe this is a regex gone wrong. All uses of Logger.warn (case insensitive) were replaced by Logger.warning :)

Yep 🤦
